functions/diff.js

```diff
@@ -17,6 +17,19 @@ const diff = (version1, version2) => {
         }
       }
     }
+    if (v1.major === v2.major &&
+        v1.minor === v2.minor &&
+        v1.patch === v2.patch && !(v1.prerelease.length && v2.prerelease.length)) {
+      const releaseTypes = []
+      for (const key in v1) {
+        if (key === 'major' || key === 'minor' || key === 'patch') {
+          if (v1[key]) {
+            releaseTypes.push(key)
+          }
+        }
+      }
+      return releaseTypes.pop()
```

Comment:
Is a minor performance change but if we do like this:

```suggestion
      const key = (v1.major && 0x001) | (v1.minor && 0x010) | (v1.patch && 0x100)

      switch (key) {
        case 0x001:
          return 'patch'

        case 0x010:
          return 'minor'

        case 0x100:
          return 'major'

        case 0x011:
          return 'patch'
        
        case 0x101:
          return 'patch'

        case 0x110:
          return 'minor'

        default:
          return undefined
      }
```

We can go from:

```
diff(0.0.1, 0.0.1-pre) x 303,099 ops/sec ±1.21% (91 runs sampled)
diff(0.0.1, 0.0.1-pre-2) x 305,103 ops/sec ±1.16% (93 runs sampled)
diff(1.1.0, 1.1.0-pre) x 306,851 ops/sec ±0.87% (94 runs sampled)
```

To:

```
diff(0.0.1, 0.0.1-pre) x 318,224 ops/sec ±1.12% (92 runs sampled)
diff(0.0.1, 0.0.1-pre-2) x 320,145 ops/sec ±0.70% (94 runs sampled)
diff(1.1.0, 1.1.0-pre) x 318,362 ops/sec ±0.94% (93 runs sampled)
```

Is a little performance improvement but is something :)

Reply:
I feel like we can get the same increase by using three `if` statements w/ early return. And it would be much more readable.

Reply:
We can change to a `map` or `hash table`, but a bunch of ifs also works.

If you change these lines to use if, also change the first ones for 3 ifs:

```js
    if (v1.major !== v2.major)
      return prefix + major

    if (v1.minor !== v2.minor)
      return prefix + minor

    if (v1.patch !== v2.patch)
      return prefix + patch
```

Reply:
I'll update this PR but leave the changes in #539 out.